x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -529,17 +529,8 @@ public QueryBuilder semanticQuery(InferenceResults inferenceResults, Integer req
                             );
                         }
 
-                        // TODO: Use WeightedTokensQueryBuilder
                         TextExpansionResults textExpansionResults = (TextExpansionResults) inferenceResults;
-                        var boolQuery = QueryBuilders.boolQuery();
-                        for (var weightedToken : textExpansionResults.getWeightedTokens()) {
-                            boolQuery.should(
-                                QueryBuilders.termQuery(inferenceResultsFieldName, weightedToken.token()).boost(weightedToken.weight())
-                            );
-                        }
-                        boolQuery.minimumShouldMatch(1);
-
-                        yield boolQuery;
+                        yield new SparseVectorQueryBuilder(name(), textExpansionResults.getWeightedTokens(), null, null, null, null);
```

Comment:
I think we need to use `inferenceResultsFieldName` here

Reply:
oups, ofc